app/src/main/java/fr/neamar/kiss/result/Result.java

```diff
@@ -88,16 +73,28 @@ else if (pojo instanceof ShortcutsPojo)
     }
 
     /**
-     * Enrich text for display. Put text requiring highlighting between {}
+     * Enrich text for display.
+     *
      *
-     * @param text to highlight
-     * @return text displayable on a textview
+     * @param text Text to highlight
+     * @param positions List of matched positions
+     * @param context Application context
+     * @return Spannable displayable on a TextView
      */
-    static Spanned enrichText(String text, Context context) {
-        //TODO: cache the result. We consume lots of CPU and RAM converting every time we display
-        return Html.fromHtml(text.replaceAll("\\{", "<font color=" + UIColors.getPrimaryColor(context) + ">").replaceAll("\\}", "</font>"));
+    static Spanned enrichText(String text, List<Pair<Integer, Integer>> positions, Context context) {
```

Comment:
Also, do we still need this function to be static?

Reply:
Not too sure why it was static in the first place.